gizmo.py

```diff
@@ -0,0 +1,90 @@
+from pandas.core.frame import DataFrame
+
+
+def hello(name,country="Finland"):
+    print("Hello {0}, how are things in {1}?".format(str(name),str(country)))
+
+def spell(input_str):
+    result=input_str[0]
+    for i in input_str[1:]:
+        result+="."+i
+        
+    return print(result)
+
+def relative_path(subject_identifiers,relpath='./subjects/moc_recording'):
+    names=[]
+    relpath='./subjects/mock_recording'
+    for i in subject_identifiers:
+        res="{0}_{1}.rec".format(relpath,i) #new style
+        #res=relpath+"_"+i+".rec" #old style, not good
+        names.append(res)
+
+    return names
+
+class Gizmo:
+    def __init__(self, name):
+        self.name=name
+
+    def speak(self):
+        print(self.name)
+
+
+def multiplication_table(zero_out_multiples=None):
+    """ 
+    Calculates a multiplication table based on two vectors ranging from 1 to 12 each.
+
+    The function multiplication_table caclulates a multiplication table for all numbers between 1 to 12. If the input argument zero_out_multiples is not None, multiplies of this input argument in the table will be all set to zero.
+
+    Parameters
+    ----------
+    zero_out_multiples: int, default = None
```

Comment:
numpydoc fails to parse this line. I think `, default = None` is not valid here. Defaults are documented in the function signature, not in here.

Reply:
I read https://numpydoc.readthedocs.io/en/latest/format.html
It says in section 4. Parameters:
If it is not necessary to specify a keyword argument, use optional: `x : int, optional
`
Is `None` not a default argument in the function?
Is it better to write default=None in the top of the function?

Reply:
Then, you need to take the documentation literal and write it as:

```
zero_out_multiples: int, optional
```

This docstring is being parsed by a program, so it needs to follow the correct syntax exactly.

Reply:
Also, there needs to be a space before the `:` character.